Lib/test/test_typing.py

```diff
@@ -3692,10 +3692,15 @@ def test_c_functions(self):
 
 
 class NewTypeTests:
+    def cleanup(self):
+        for f in self.module._cleanups:
+            f()
```

Comment:
Those seem to be cleaning LRU caches for various type `__getitem__`.

Reply:
Yup, and it's clear `Union.__getitem__` cache which lead to memory leak.